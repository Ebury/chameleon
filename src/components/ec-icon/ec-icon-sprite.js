import config from '../../config';

const spriteSourceUrls = new Map([
  ['rounded', '/rounded-icons.svg'],
  ['simple', '/simple-icons.svg'],
  ['currency', '/currency-flags.svg'],
]);

export function getSpriteSourceByName(iconFullName) {
  /* istanbul ignore if */
  if (!iconFullName) {
    throw new Error('Missing iconFullName');
  }

  const [spriteName, iconName] = iconFullName.split('-');
  if (!spriteName || !iconName) {
    throw new Error(`Unable to determine the source of SVG sprite for icon with name: ${iconFullName}`);
  }
  return getSpriteUrl(spriteName);
}

function getSpriteUrl(spriteName) {
  const spriteUrl = spriteSourceUrls.get(spriteName);
  if (!spriteUrl) {
    throw new Error(`The sprite "${spriteName}" doesn't exist.`);
  }

  return `${config.iconsPublicPath}${spriteUrl}`;
}
