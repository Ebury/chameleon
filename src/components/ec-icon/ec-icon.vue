<template>
  <svg
    :width="size"
    :height="size"
    :class="['ec-icon', typeClass]"
  >
    <use :xlink:href="`#ec-${name}`" />
  </svg>
</template>

<script>
export default {
  name: 'EcIcon',
  props: {
    name: {
      type: String,
      required: true,
    },
    size: {
      type: Number,
    },
    type: {
      type: String,
      validator(value) {
        return ['error', 'info', 'success', 'warning'].includes(value);
      },
    },
  },
  computed: {
    typeClass() {
      return this.type ? `ec-icon--${this.type}` : null;
    },
  },
};
</script>

<style lang="scss">
@import '../../scss/settings/colors/reserved-color-scale';

.ec-icon {
  &--error {
    fill: $level-2-error;
  }

  &--info {
    fill: $level-2-info;
  }

  &--success {
    fill: $level-2-success;
  }

  &--warning {
    fill: $level-2-warning;
  }
}
</style>
