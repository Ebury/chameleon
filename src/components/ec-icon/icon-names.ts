// This code is generated by build-types.js: do not change it manually

export enum IconName {
  CURRENCY_AED = 'currency-aed',
  CURRENCY_ANG = 'currency-ang',
  CURRENCY_AUD = 'currency-aud',
  CURRENCY_BBD = 'currency-bbd',
  CURRENCY_BGN = 'currency-bgn',
  CURRENCY_BHD = 'currency-bhd',
  CURRENCY_BRL = 'currency-brl',
  CURRENCY_BSD = 'currency-bsd',
  CURRENCY_BWP = 'currency-bwp',
  CURRENCY_BZD = 'currency-bzd',
  CURRENCY_CAD = 'currency-cad',
  CURRENCY_CDF = 'currency-cdf',
  CURRENCY_CHF = 'currency-chf',
  CURRENCY_CNY = 'currency-cny',
  CURRENCY_CZK = 'currency-czk',
  CURRENCY_DKK = 'currency-dkk',
  CURRENCY_EGP = 'currency-egp',
  CURRENCY_EUR = 'currency-eur',
  CURRENCY_FJD = 'currency-fjd',
  CURRENCY_GBP = 'currency-gbp',
  CURRENCY_GHS = 'currency-ghs',
  CURRENCY_GNF = 'currency-gnf',
  CURRENCY_HKD = 'currency-hkd',
  CURRENCY_HRK = 'currency-hrk',
  CURRENCY_HUF = 'currency-huf',
  CURRENCY_ILS = 'currency-ils',
  CURRENCY_INR = 'currency-inr',
  CURRENCY_ISK = 'currency-isk',
  CURRENCY_JMD = 'currency-jmd',
  CURRENCY_JOD = 'currency-jod',
  CURRENCY_JPY = 'currency-jpy',
  CURRENCY_KES = 'currency-kes',
  CURRENCY_KWD = 'currency-kwd',
  CURRENCY_KYD = 'currency-kyd',
  CURRENCY_KZT = 'currency-kzt',
  CURRENCY_LKR = 'currency-lkr',
  CURRENCY_LRD = 'currency-lrd',
  CURRENCY_LSL = 'currency-lsl',
  CURRENCY_MGA = 'currency-mga',
  CURRENCY_MUR = 'currency-mur',
  CURRENCY_MWK = 'currency-mwk',
  CURRENCY_MXN = 'currency-mxn',
  CURRENCY_MZN = 'currency-mzn',
  CURRENCY_NAD = 'currency-nad',
  CURRENCY_NGN = 'currency-ngn',
  CURRENCY_NO_FLAG = 'currency-no-flag',
  CURRENCY_NOK = 'currency-nok',
  CURRENCY_NZD = 'currency-nzd',
  CURRENCY_OMR = 'currency-omr',
  CURRENCY_PHP = 'currency-php',
  CURRENCY_PKR = 'currency-pkr',
  CURRENCY_PLN = 'currency-pln',
  CURRENCY_QAR = 'currency-qar',
  CURRENCY_RON = 'currency-ron',
  CURRENCY_RSD = 'currency-rsd',
  CURRENCY_RUB = 'currency-rub',
  CURRENCY_RWF = 'currency-rwf',
  CURRENCY_SAR = 'currency-sar',
  CURRENCY_SEK = 'currency-sek',
  CURRENCY_SGD = 'currency-sgd',
  CURRENCY_SLL = 'currency-sll',
  CURRENCY_SZL = 'currency-szl',
  CURRENCY_THB = 'currency-thb',
  CURRENCY_TND = 'currency-tnd',
  CURRENCY_TRY = 'currency-try',
  CURRENCY_TTD = 'currency-ttd',
  CURRENCY_TZS = 'currency-tzs',
  CURRENCY_UGX = 'currency-ugx',
  CURRENCY_USD = 'currency-usd',
  CURRENCY_XAF = 'currency-xaf',
  CURRENCY_XCD = 'currency-xcd',
  CURRENCY_ZAR = 'currency-zar',
  CURRENCY_ZMW = 'currency-zmw',
  ROUNDED_BOOKS = 'rounded-books',
  ROUNDED_CALENDAR = 'rounded-calendar',
  ROUNDED_CANCELLED = 'rounded-cancelled',
  ROUNDED_CHECK = 'rounded-check',
  ROUNDED_DOCUMENTS = 'rounded-documents',
  ROUNDED_EURO = 'rounded-euro',
  ROUNDED_HOURGLASS = 'rounded-hourglass',
  ROUNDED_INFO = 'rounded-info',
  ROUNDED_NOTIFICATION = 'rounded-notification',
  ROUNDED_OPEN = 'rounded-open',
  ROUNDED_PARTIAL = 'rounded-partial',
  ROUNDED_PAUSE = 'rounded-pause',
  ROUNDED_REJECTED = 'rounded-rejected',
  ROUNDED_RETURNED = 'rounded-returned',
  ROUNDED_SETTINGS = 'rounded-settings',
  ROUNDED_THUMB_DOWN = 'rounded-thumb-down',
  ROUNDED_THUMB_UP_DOWN = 'rounded-thumb-up-down',
  ROUNDED_THUMB_UP = 'rounded-thumb-up',
  ROUNDED_TRADE = 'rounded-trade',
  ROUNDED_UNKNOWN = 'rounded-unknown',
  ROUNDED_VERIFIED = 'rounded-verified',
  SIMPLE_ACCOUNTS = 'simple-accounts',
  SIMPLE_ADD = 'simple-add',
  SIMPLE_ADDRESS = 'simple-address',
  SIMPLE_ALIAS = 'simple-alias',
  SIMPLE_AMOUNT = 'simple-amount',
  SIMPLE_ARROW_DROP_DOWN = 'simple-arrow-drop-down',
  SIMPLE_ARROW_LEFT = 'simple-arrow-left',
  SIMPLE_ARROW_RIGHT = 'simple-arrow-right',
  SIMPLE_ARROW_UP_DOWN = 'simple-arrow-up-down',
  SIMPLE_ATTACHMENT = 'simple-attachment',
  SIMPLE_BANK = 'simple-bank',
  SIMPLE_BATCH_PAYMENT = 'simple-batch-payment',
  SIMPLE_BENEFICIARIES = 'simple-beneficiaries',
  SIMPLE_BLOCK = 'simple-block',
  SIMPLE_BUY = 'simple-buy',
  SIMPLE_CALENDAR = 'simple-calendar',
  SIMPLE_CHECK = 'simple-check',
  SIMPLE_CHEVRON_DOWN = 'simple-chevron-down',
  SIMPLE_CHEVRON_LEFT = 'simple-chevron-left',
  SIMPLE_CHEVRON_RIGHT = 'simple-chevron-right',
  SIMPLE_CHEVRON_UP = 'simple-chevron-up',
  SIMPLE_CITY = 'simple-city',
  SIMPLE_CLOSE = 'simple-close',
  SIMPLE_CLOUD_UPLOAD = 'simple-cloud-upload',
  SIMPLE_COLLECT = 'simple-collect',
  SIMPLE_CONVERT_AND_PAY = 'simple-convert-and-pay',
  SIMPLE_COPY = 'simple-copy',
  SIMPLE_COUNTRY = 'simple-country',
  SIMPLE_CURRENCY = 'simple-currency',
  SIMPLE_DASHBOARD_REVERSE = 'simple-dashboard-reverse',
  SIMPLE_DASHBOARD = 'simple-dashboard',
  SIMPLE_DELETE = 'simple-delete',
  SIMPLE_DOCUMENT = 'simple-document',
  SIMPLE_DOWNLOAD = 'simple-download',
  SIMPLE_DRAWDOWN = 'simple-drawdown',
  SIMPLE_EDIT = 'simple-edit',
  SIMPLE_ERROR = 'simple-error',
  SIMPLE_EYE = 'simple-eye',
  SIMPLE_FEE = 'simple-fee',
  SIMPLE_HEDGING = 'simple-hedging',
  SIMPLE_HELP = 'simple-help',
  SIMPLE_INCOMING_FUNDS = 'simple-incoming-funds',
  SIMPLE_INFO = 'simple-info',
  SIMPLE_LOADING = 'simple-loading',
  SIMPLE_LOCKER = 'simple-locker',
  SIMPLE_MAIL = 'simple-mail',
  SIMPLE_MENU = 'simple-menu',
  SIMPLE_MORE = 'simple-more',
  SIMPLE_NUMBERS = 'simple-numbers',
  SIMPLE_OUTLINE_DELETE = 'simple-outline-delete',
  SIMPLE_PAYMENT_LIGHT = 'simple-payment-light',
  SIMPLE_PAYMENT = 'simple-payment',
  SIMPLE_PERSON = 'simple-person',
  SIMPLE_POSTAL_CODE = 'simple-postal-code',
  SIMPLE_REASON = 'simple-reason',
  SIMPLE_REFRESH = 'simple-refresh',
  SIMPLE_RELATIONSHIP = 'simple-relationship',
  SIMPLE_REQUEST = 'simple-request',
  SIMPLE_REUSE = 'simple-reuse',
  SIMPLE_SEARCH = 'simple-search',
  SIMPLE_SELL = 'simple-sell',
  SIMPLE_SIGN_OUT = 'simple-sign-out',
  SIMPLE_SNOWFLAKE = 'simple-snowflake',
  SIMPLE_THUMB_DOWN = 'simple-thumb-down',
  SIMPLE_THUMB_UP = 'simple-thumb-up',
  SIMPLE_TRADE_FINANCE_WITH_UNDERLINE = 'simple-trade-finance-with-underline',
  SIMPLE_TRADE_FINANCE = 'simple-trade-finance',
  SIMPLE_TRADE = 'simple-trade',
  SIMPLE_TRANSACTION = 'simple-transaction',
  SIMPLE_WALLET = 'simple-wallet',
  SIMPLE_WAND = 'simple-wand',
  SIMPLE_WITHDRAW = 'simple-withdraw',
}
