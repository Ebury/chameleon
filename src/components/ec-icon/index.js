export { default } from './ec-icon.vue';
export * as IconName from './iconNames';
