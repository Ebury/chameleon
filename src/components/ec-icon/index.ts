export { default } from './ec-icon.vue';
export * as types from './types';
