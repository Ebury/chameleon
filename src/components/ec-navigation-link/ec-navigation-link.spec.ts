import { type ComponentMountingOptions, mount } from '@vue/test-utils';
import { vi } from 'vitest';

import { IconName } from '../ec-icon/icon-names';
import EcNavigationLink from './ec-navigation-link.vue';
import type { NavigationLinkProps } from './types';

describe('EcNavigationLink', () => {
  describe('as router-link', () => {
    function mountAsRouterLink(props?: Partial<NavigationLinkProps>, mountOpts?: ComponentMountingOptions<typeof EcNavigationLink>) {
      return mount(EcNavigationLink, {
        props: {
          text: 'Link',
          iconName: IconName.SIMPLE_CHECK,
          url: '/balances',
          isRouterLink: true,
          ...props,
        },
        ...mountOpts,
      });
    }

    it('should render as a router link if isRouterLink prop set to true', () => {
      const wrapper = mountAsRouterLink({ isRouterLink: true });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should be expanded by default', () => {
      const wrapper = mountAsRouterLink();
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link').classes('ec-navigation-link--is-collapsed')).toBe(false);
    });

    it('should hide the text when is collapsed', () => {
      const wrapper = mountAsRouterLink({ isCollapsed: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(false);
    });

    it('should show the text when is expanded', () => {
      const wrapper = mountAsRouterLink({ isCollapsed: false });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(true);
    });

    it('should be active when isActive is passed into', () => {
      const wrapper = mountAsRouterLink({ isActive: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link').classes('ec-navigation-link--is-active')).toBe(true);
    });

    it('should be compact when isCompact is passed into', () => {
      const wrapper = mountAsRouterLink({ isCompact: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link').classes('ec-navigation-link--is-compact')).toBe(true);
    });

    it('should pass listeners from parent to the root', () => {
      const clickSpy = vi.fn();
      const wrapper = mountAsRouterLink({}, {
        attrs: { onClick: clickSpy },
      });

      wrapper.findByDataTest('ec-navigation-link').trigger('click');
      expect(clickSpy).toHaveBeenCalledTimes(1);
    });

    it('should pass custom attributes', () => {
      const wrapper = mountAsRouterLink({}, {
        attrs: {
          id: 'my-link',
          'data-test': 'my-custom-link',
          class: 'my-custom-class',
        },
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('as regular anchor', () => {
    function mountAsAnchor(props?: Partial<NavigationLinkProps>, mountOpts?: ComponentMountingOptions<typeof EcNavigationLink>) {
      return mount(EcNavigationLink, {
        props: {
          text: 'Link',
          iconName: IconName.SIMPLE_CHECK,
          url: '#/balances',
          isRouterLink: false,
          ...props,
        },
        ...mountOpts,
      });
    }

    it('should render as a normal <a> tag if isRouterLink is set to false', () => {
      const wrapper = mountAsAnchor({ isRouterLink: false });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should be expanded by default', () => {
      const wrapper = mountAsAnchor();
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link').classes('ec-navigation-link--is-collapsed')).toBe(false);
    });

    it('should hide the text when is collapsed', () => {
      const wrapper = mountAsAnchor({ isCollapsed: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(false);
    });

    it('should show the text when is expanded', () => {
      const wrapper = mountAsAnchor({ isCollapsed: false });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link__text').isVisible()).toBe(true);
    });

    it('should be active when isActive is passed into', () => {
      const wrapper = mountAsAnchor({ isActive: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link').classes('ec-navigation-link--is-active')).toBe(true);
    });

    it('should be compact when isCompact is passed into', () => {
      const wrapper = mountAsAnchor({ isCompact: true });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-navigation-link').classes('ec-navigation-link--is-compact')).toBe(true);
    });

    it('should pass listeners from parent to the root', () => {
      const clickSpy = vi.fn();
      const wrapper = mountAsAnchor({}, {
        attrs: { onClick: clickSpy },
      });

      wrapper.findByDataTest('ec-navigation-link').trigger('click');
      expect(clickSpy).toHaveBeenCalledTimes(1);
    });

    it('should pass custom attributes', () => {
      const wrapper = mountAsAnchor({}, {
        attrs: {
          id: 'my-link',
          'data-test': 'my-custom-link',
          class: 'my-custom-class',
        },
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
