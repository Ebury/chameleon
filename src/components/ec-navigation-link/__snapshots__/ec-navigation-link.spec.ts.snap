// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcNavigationLink > as regular anchor > should be active when isActive is passed into 1`] = `
<a
  class="ec-navigation-link ec-navigation-link--is-active"
  data-test="ec-navigation-link"
  href="#/balances"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as regular anchor > should be compact when isCompact is passed into 1`] = `
<a
  class="ec-navigation-link ec-navigation-link--is-compact"
  data-test="ec-navigation-link"
  href="#/balances"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as regular anchor > should be expanded by default 1`] = `
<a
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  href="#/balances"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as regular anchor > should hide the text when is collapsed 1`] = `
<a
  class="ec-navigation-link ec-navigation-link--is-collapsed"
  data-test="ec-navigation-link"
  href="#/balances"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
      style="display: none;"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as regular anchor > should pass custom attributes 1`] = `
<a
  class="my-custom-class ec-navigation-link"
  data-test="my-custom-link ec-navigation-link"
  href="#/balances"
  id="my-link"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as regular anchor > should render as a normal <a> tag if isRouterLink is set to false 1`] = `
<a
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  href="#/balances"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as regular anchor > should show the text when is expanded 1`] = `
<a
  class="ec-navigation-link"
  data-test="ec-navigation-link"
  href="#/balances"
>
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
</a>
`;

exports[`EcNavigationLink > as router-link > should be active when isActive is passed into 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link ec-navigation-link--is-active"
  data-test="ec-navigation-link ec-stub router-link-stub"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;

exports[`EcNavigationLink > as router-link > should be compact when isCompact is passed into 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link ec-navigation-link--is-compact"
  data-test="ec-navigation-link ec-stub router-link-stub"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;

exports[`EcNavigationLink > as router-link > should be expanded by default 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link"
  data-test="ec-navigation-link ec-stub router-link-stub"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;

exports[`EcNavigationLink > as router-link > should hide the text when is collapsed 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link ec-navigation-link--is-collapsed"
  data-test="ec-navigation-link ec-stub router-link-stub"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
      style="display: none;"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;

exports[`EcNavigationLink > as router-link > should pass custom attributes 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="my-custom-class ec-navigation-link"
  data-test="my-custom-link ec-navigation-link ec-stub router-link-stub"
  id="my-link"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;

exports[`EcNavigationLink > as router-link > should render as a router link if isRouterLink prop set to true 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link"
  data-test="ec-navigation-link ec-stub router-link-stub"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;

exports[`EcNavigationLink > as router-link > should show the text when is expanded 1`] = `
<ec-stub
  active-class="ec-navigation-link--is-active"
  class="ec-navigation-link"
  data-test="ec-navigation-link ec-stub router-link-stub"
  to="/balances"
>
  
  <svg
    class="ec-icon ec-navigation-link__icon"
    data-test="ec-navigation-link__icon"
    height="24"
    width="24"
  >
    <use
      xlink:href="#ec-simple-check"
    />
  </svg>
  <transition-stub
    appear="false"
    css="true"
    name="ec-navigation-link__text-fade"
    persisted="true"
  >
    <span
      class="ec-navigation-link__text"
      data-test="ec-navigation-link__text"
    >
      Link
    </span>
  </transition-stub>
  
</ec-stub>
`;
